src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -363,6 +364,18 @@ int sd_dhcp6_client_set_request_option(sd_dhcp6_client *client, uint16_t option)
         return 0;
 }
 
+int sd_dhcp6_client_set_request_mud_url(sd_dhcp6_client *client, char *mudurl) {
+
+        assert_return(client, -EINVAL);
+        assert_return(client->state == DHCP6_STATE_STOPPED, -EBUSY);
+        assert_return(mudurl, -EINVAL);
+        assert_return(strlen(mudurl) <= 255, -EINVAL);
```

Comment:
... and add a call to http_url_is_valid() here. (Unless we want to pass through invalid URLs for some reason, but I don't think so.)

Reply:
I think @keszybz's points hold true for the other PRs as well.  Nice catch.